### PR TITLE
fix(wash-cli): developer loop improvements

### PR DESCRIPTION
**Pull request by brooksmtownsend**
- **fix(wash-cli)!: remove duplicate components, ignore generated paths**
- **fix(wash-dev): combine links**
- **fix(wash-cli): continue dev loop when build fails**
- **fix(wash-cli): differentiate config, support fast-reload**

## Feature or Problem
This PR includes a couple more fixes for the stability + capabilities of wash dev, separated by commits

### Manual Verification

Signed-off-by: Brooks Townsend <brooksmtownsend@gmail.com>

### CODE REVIEW

File: crates/wash-cli/src/dev.rs [hunk 1]
```diff
@@ -799,7 +798,7 @@ impl ProjectDeps {
             metadata: Metadata {
                 name: app_name,
                 annotations: BTreeMap::from([("version".into(), "v0.0.0".into())]),
-                labels: BTreeMap::from([("wasmcloud.dev/generated-by".into(), "wash dev".into())]),
+                labels: BTreeMap::from([("wasmcloud.dev/generated-by".into(), "wash-dev".into())]),
```
Comment (lachieh) [inline crates/wash-cli/src/dev.rs:801]: good to include wash version in here?

Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:801]: Good idea!

Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:801]: Added based on the crate version 😄

File: crates/wash-cli/src/dev.rs [hunk 2]
```diff
@@ -1527,22 +1534,35 @@ async fn run_dev_loop(
...
         style("Building project...").bold(),
     );
-    let artifact_path = build_project(project_cfg, Some(&SignConfig::default()))
+    let Ok(Ok(artifact_path)) = build_project(project_cfg, Some(&SignConfig::default()))
         .await
-        .context("failed to build project")?
-        .canonicalize()
-        .context("failed to canonicalize path")?;
+        .context("failed to build project")
+        .map(|p| p.canonicalize().context("failed to canonicalize path"))
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/dev.rs:1540]: I *wonder* if it's worth putting *which* path in here.. Not sure the canonicalize error includes the path it was actually trying to canonicalize

File: crates/wash-cli/src/dev.rs [hunk 3]
```diff
@@ -1527,22 +1534,35 @@ async fn run_dev_loop(
...
 
     // After the project is built, we must ensure dependencies are set up and running
     let (resolve, world_id) = if let TypeConfig::Component(_) = project_cfg.project_type {
-        let component_bytes = tokio::fs::read(&artifact_path)
-            .await
-            .context("failed to read component bytes from built artifact path")?;
+        let component_bytes = tokio::fs::read(&artifact_path).await.with_context(|| {
+            format!(
+                "failed to read component bytes from built artifact path {}",
+                artifact_path.to_string_lossy()
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/dev.rs:1563]:
```suggestion
                "failed to read component bytes from built artifact path [{}]",
                artifact_path.display()
```

Can we not use `display()` here?

File: crates/wash-cli/src/dev.rs [hunk 4]
```diff
@@ -1587,12 +1608,13 @@ async fn run_dev_loop(
     );
     let component = generate_component_from_project_cfg(project_cfg, &component_id, &component_ref)
         .context("failed to generate app component")?;
+    current_project_deps.component = Some(component);
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/dev.rs:1611]: we move this in here, so we can probably change the line above it right?

File: crates/wash-cli/src/dev.rs [hunk 5]
```diff
@@ -1900,7 +1929,7 @@ pub async fn handle_command(
             _ => {}
         },
         Err(e) => {
-            eprintln!("[error] watch failed: {:?}", e);
+            eprintln!("{} [error] watch failed: {:?}", emoji::ERROR, e);
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/dev.rs:1932]:
```suggestion
            eprintln!("{} Watch failed: {:?}", emoji::ERROR, e);
```

File: crates/wash-cli/src/dev.rs [hunk 6]
```diff
@@ -2015,9 +2044,11 @@ pub async fn handle_command(
 
                     // Stop NATS
                     if let Some(mut nats) = nats_child {
+                        eprintln!("{} Stopping NATS...", emoji::HOURGLASS_DRAINING);
                         nats.kill().await?;
                     }
                 }
+                eprint!("✅ Dev session exited successfully");
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/dev.rs:2051]: We don't want the newline here? But at lease we can use something from `emoji::`

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket: , nothing but nits

File: crates/wash-cli/src/dev.rs (hunk 3 above)
Comment (brooksmtownsend) [reply crates/wash-cli/src/dev.rs:1563]: I didn't know that existed! Neat

**Pull request merged**